packages/web-ui/src/utils/with-breakpoints.ts

```diff
@@ -0,0 +1,20 @@
+import { Breakpoints, Responsive } from '../types';
+import { getPrefixedName } from './utils';
+
+export const withBreakpoints = (value: Responsive<string> | undefined, prefix = '') => {
+  if (typeof value === 'string') {
+    return getPrefixedName(`${prefix}-${value}`);
+  }
+
+  const classes: string[] = [];
+  if (typeof value === 'object') {
+    for (const bp of Object.keys(value) as Breakpoints[]) {
```

Comment:
Probably better to flip this logic to return sooner, or at least move the `classes` declaration inside the condition it relates to.

```js
if (typeof value !== 'object') {
  return;
}

const classes: string[] = [];
for (...) {
  ...
}
```